include/deal.II/grid/grid_tools.h

```diff
@@ -1060,25 +1057,29 @@ namespace GridTools
    * orthogonal_equality().
    *
    * The bitset that is returned inside of PeriodicFacePair encodes the
-   * _relative_ orientation of the first face with respect to the second face,
-   * see the documentation of orthogonal_equality() for further details.
+   * _relative_ orientation of the first face with respect to the second
+   * face, see the documentation of orthogonal_equality() for further
+   * details.
    *
    * The @p direction refers to the space direction in which periodicity is
-   * enforced.
+   * enforced. When maching periodic faces this vector component is ignored.
    *
    * The @p offset is a vector tangential to the faces that is added to the
    * location of vertices of the 'first' boundary when attempting to match
-   * them to the corresponding vertices of the 'second' boundary. This can be
-   * used to implement conditions such as $u(0,y)=u(1,y+1)$.
-   *
-   * Optionally a (dim x dim) rotation matrix @p matrix along with a vector @p
-   * first_vector_components can be specified that describes how vector valued
-   * DoFs of the first face should be modified prior to constraining to the
-   * DoFs of the second face. If @p first_vector_components is non empty the
-   * matrix is interpreted as a rotation matrix that is applied to all vector
-   * valued blocks listed in @p first_vector_components of the FESystem. For
-   * more details see make_periodicity_constraints() and the glossary
-   * @ref GlossPeriodicConstraints "glossary entry on periodic conditions".
+   * them to the corresponding vertices of the 'second' boundary. This can
+   * be used to implement conditions such as $u(0,y)=u(1,y+1)$.
+   *
+   * Optionally, a @p matrix can be specified that describes how vector
+   * valued DoFs of the first face should be modified prior to constraining
+   * to the DoFs of the second face. If the matrix has size
+   * $n\_face\_dofs\times n\_face\_dofs$, the periodicity constraints are
+   * applied as $dofs\_2 = matrix\cdot dofs\_1$. If the matrix has size
+   * $dim\times dim$, the matrix is interpreted as a rotation matrix for
+   * vector valued components.
+   * For more details see DoFTools::make_periodicity_constraints(), the
```

Comment:
But then I don't know what the matrix argument does. I suppose that it doesn't actually enter the algorithm that determines what is or isn't a matching face pair, but that you simply store the matrix in the PeriodicFacePair, right? I think it would be worthwhile stating this here: that the argument isn't actually interpreted or evaluated, but simply passed down into the returned object for later use. This would also free you from having to explain how it is interpreted and state that that is the job of the function that receives the list.